Init Audio Engine Pull

maxbolton opened the pull request:
- Introduces the Audio Engine, Audio Object and its derived classes as components (Audio Source & Listener).
> Provides basic methods for loading, playing & stopping a sound
> Instantiation method to init static engine object
> static method to convert <vector> to FMOD_VECTOR

- Modifies CMake files for the overarching project to create a ThirdParty directory (Instructions of FMOD installation to be provided)
- Removes Old CSC8503 build files

Commit:
Audio Engine init commit.
Basic audio object classes and methods outlined/implemented. Cmake updated to use necessary libraries and setup ThirdParty folder for middleware (FMOD)

Commit:
Remove Old build files

Commit:
Merge branch 'main' into AudioEngine-impl

Commit:
Modifiy AudioObjects to use Icomponent base class. Modified child class and file names accordingly.

Alasdair-PB commented on AudioEngine/AudioSourceComponent.h:
Would it be possible to multiply the local Component volume by a global volume so we can modify volume within settings in the future.

maxbolton replied:
A master audio group has been added to the audio engine. Each individual object will be added to this group for now, in the future more groups can be added for specific types of sounds (FX, Music, Chat, etc). Each object will still have its own individual volume.

Commit:
Add master volume control

Added setMasterVolume and getMasterGroup methods to AudioEngine.
Introduced masterGroup member variable in AudioEngine.
Modified AudioObject to store a pointer to AudioEngine.
Updated AudioSourceComponent to use master audio group.

Alasdair-PB reviewed (approved):
Looks good to me

mlhumphriss reviewed (approved):
Yeah read through it and looks good, no obvious errors
